A005 and A002 both trigger for lambda's

This example triggers two violations (with preview enabled), is this intentional?

```python
lambda id: id + 1
```

```
$ ruff check --isolated --preview --select A test.py

test.py:1:8: A002 Argument `id` is shadowing a Python builtin
  |
1 | lambda id: id + 1
  |        ^^ A002
  |

test.py:1:8: A006 Lambda argument `id` is shadowing a Python builtin
  |
1 | lambda id: id + 1
  |        ^^ A006
  |
```

```
$ ruff --version
ruff 0.7.2
```

--- a/crates/ruff_linter/resources/test/fixtures/flake8_builtins/A002.py
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_builtins/A002.py
@@ -10,6 +10,8 @@ async def func3(id, dir):
     pass
 
 
+# this is Ok for A002 (trigger A005 instead)
+# https://github.com/astral-sh/ruff/issues/14135
 map([], lambda float: ...)
 
 from typing import override, overload

--- a/crates/ruff_linter/resources/test/fixtures/flake8_builtins/A006.py
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_builtins/A006.py
@@ -3,3 +3,8 @@ lambda x, float, y: x + y
 lambda min, max: min
 lambda id: id
 lambda dir: dir
+
+# Ok for A006 - should trigger A002 instead
+# https://github.com/astral-sh/ruff/issues/14135
+def func1(str, /, type, *complex, Exception, **getattr):
+    pass

--- a/crates/ruff_linter/src/rules/flake8_builtins/rules/builtin_argument_shadowing.rs
+++ b/crates/ruff_linter/src/rules/flake8_builtins/rules/builtin_argument_shadowing.rs
@@ -1,7 +1,7 @@
 use ruff_diagnostics::Diagnostic;
 use ruff_diagnostics::Violation;
 use ruff_macros::{derive_message_formats, violation};
-use ruff_python_ast::Parameter;
+use ruff_python_ast::{Expr, Parameter};
 use ruff_python_semantic::analyze::visibility::{is_overload, is_override};
 use ruff_text_size::Ranged;
 
@@ -70,6 +70,15 @@ pub(crate) fn builtin_argument_shadowing(checker: &mut Checker, parameter: &Para
         &checker.settings.flake8_builtins.builtins_ignorelist,
         checker.settings.target_version,
     ) {
+        // Ignore parameters in lambda expressions.
+        // (That is the domain of A006.)
+        if checker
+            .semantic()
+            .current_expression()
+            .is_some_and(Expr::is_lambda_expr)
+        {
+            return;
+        }
         // Ignore `@override` and `@overload` decorated functions.
         if checker
             .semantic()

--- a/crates/ruff_linter/src/rules/flake8_builtins/rules/builtin_argument_shadowing.rs
+++ b/crates/ruff_linter/src/rules/flake8_builtins/rules/builtin_argument_shadowing.rs
@@ -58,7 +58,7 @@ impl Violation for BuiltinArgumentShadowing {
     #[derive_message_formats]
     fn message(&self) -> String {
         let BuiltinArgumentShadowing { name } = self;
-        format!("Argument `{name}` is shadowing a Python builtin")
+        format!("Function argument `{name}` is shadowing a Python builtin")
     }
 }
 

